src/js/modules/Product.js

```diff
@@ -0,0 +1,111 @@
+import { template } from "./product-template.js";
+import Wish from "./wish.js";
+import ProductServices from "../services/sharedproduct.service.js";
+
+class Product {
+  startIndex = 0;
+  pageSize = 3;
+  currentProducts = [];
+  productListElement = document.getElementById("product-list");
+  numberProducts = 0;
+  numberProductsContainer = document.getElementById("numberProducts");
+
+  constructor() {
+    this.loadProducts();
+    this.initEvents();
+  }
+
+  loadProducts = async () => {
+    try {
+      await ProductServices.fetchProducts();
+      this.numberProductsContainer.innerHTML = ProductServices.productsCount;
+      this.loadNewProductsToRender();
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  displayProducts(products = []) {
+    let existingCart = ProductServices.getCartProducts();
+  
+    const promise = new Promise((resolve, reject) => {
```

Comment:
Try `callback` func

Reply:
should replace async-await with promise for now